### ISSUE TITLE
Failed to acquire global channel lock

### ISSUE THREAD

**Issue by Sarisan** [microsoft/WSL#9033]
### Version

Windows version: 10.0.22621.674

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.68.1

### Distro Version

Debian GNU/Linux bookworm/sid, Arch Linux

### Repro Steps

Just run WSL as always

### Expected Behavior

To see no errors

### Actual Behavior

Failed to mount D:\, see dmesg for more details.

### Diagnostic Logs

```
[    0.000000] Linux version 6.0.1-microsoft-quark-WSL2 (sarisan@GrayFox) (gcc (Debian 12.2.0-5) 12.2.0, GNU ld (GNU Binutils for Debian) 2.39) #1 SMP PREEMPT_DYNAMIC Wed Oct 12 18:49:24 JST 2022
[    0.000000] Command line: initrd=\initrd.img WSL_ROOT_INIT=1 panic=-1 nr_cpus=4 swiotlb=force console=hvc0 debug pty.legacy_count=0
[    0.000000] KERNEL supported cpus:
[    0.000000]   Intel GenuineIntel
[    0.000000]   AMD AuthenticAMD
[    0.000000]   Centaur CentaurHauls
[    0.000000] x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
[    0.000000] x86/fpu: Supporting XSAVE feature 0x002: 'SSE registers'
[    0.000000] x86/fpu: Enabled xstate features 0x3, context size is 576 bytes, using 'compacted' format.
[    0.000000] signal: max sigframe size: 1520
[    0.000000] BIOS-provided physical RAM map:
[    0.000000] BIOS-e820: [mem 0x0000000000000000-0x000000000009ffff] usable
[    0.000000] BIOS-e820: [mem 0x00000000000e0000-0x00000000000e0fff] reserved
[    0.000000] BIOS-e820: [mem 0x0000000000100000-0x00000000001fffff] ACPI data
[    0.000000] BIOS-e820: [mem 0x0000000000200000-0x000000007affffff] usable
[    0.000000] NX (Execute Disable) protection: active
[    0.000000] DMI not present or invalid.
[    0.000000] Hypervisor detected: Microsoft Hyper-V
[    0.000000] Hyper-V: privilege flags low 0x2e7f, high 0x3b8030, hints 0x66c2c, misc 0xe4bed7b6
[    0.000000] Hyper-V: Host Build 10.0.22621.674-0-0
[    0.000000] Hyper-V: Nested features: 0x3e0101
[    0.000000] Hyper-V: LAPIC Timer Frequency: 0x1e8480
[    0.000000] Hyper-V: Using hypercall for remote TLB flush
[    0.000000] clocksource: hyperv_clocksource_tsc_page: mask: 0xffffffffffffffff max_cycles: 0x24e6a1710, max_idle_ns: 440795202120 ns
[    0.000007] tsc: Marking TSC unstable due to running on Hyper-V
[    0.000014] tsc: Detected 1094.399 MHz processor
[    0.000049] e820: update [mem 0x00000000-0x00000fff] usable ==> reserved
[    0.000057] e820: remove [mem 0x000a0000-0x000fffff] usable
[    0.000065] last_pfn = 0x7b000 max_arch_pfn = 0x400000000
[    0.000148] x86/PAT: Configuration [0-7]: WB  WC  UC- UC  WB  WP  UC- WT
[    0.000214] Using GB pages for direct mapping
[    0.000768] RAMDISK: [mem 0x03835000-0x03a0cfff]
[    0.000781] ACPI: Early table checksum verification disabled
[    0.000829] ACPI: RSDP 0x00000000000E0000 000024 (v02 VRTUAL)
[    0.000841] ACPI: XSDT 0x0000000000100000 000044 (v01 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000854] ACPI: FACP 0x0000000000101000 000114 (v06 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000867] ACPI: DSDT 0x00000000001011B8 01E191 (v02 MSFTVM DSDT01   00000001 MSFT 05000000)
[    0.000877] ACPI: FACS 0x0000000000101114 000040
[    0.000885] ACPI: OEM0 0x0000000000101154 000064 (v01 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000894] ACPI: SRAT 0x000000000011F349 0002F0 (v02 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000903] ACPI: APIC 0x000000000011F639 000068 (v04 VRTUAL MICROSFT 00000001 MSFT 00000001)
[    0.000910] ACPI: Reserving FACP table memory at [mem 0x101000-0x101113]
[    0.000914] ACPI: Reserving DSDT table memory at [mem 0x1011b8-0x11f348]
[    0.000916] ACPI: Reserving FACS table memory at [mem 0x101114-0x101153]
[    0.000918] ACPI: Reserving OEM0 table memory at [mem 0x101154-0x1011b7]
[    0.000920] ACPI: Reserving SRAT table memory at [mem 0x11f349-0x11f638]
[    0.000922] ACPI: Reserving APIC table memory at [mem 0x11f639-0x11f6a0]
[    0.013303] Zone ranges:
[    0.013311]   DMA      [mem 0x0000000000001000-0x0000000000ffffff]
[    0.013318]   DMA32    [mem 0x0000000001000000-0x000000007affffff]
[    0.013321]   Normal   empty
[    0.013323]   Device   empty
[    0.013325] Movable zone start for each node
[    0.013326] Early memory node ranges
[    0.013327]   node   0: [mem 0x0000000000001000-0x000000000009ffff]
[    0.013330]   node   0: [mem 0x0000000000200000-0x000000007affffff]
[    0.013333] Initmem setup node 0 [mem 0x0000000000001000-0x000000007affffff]
[    0.016728] On node 0, zone DMA: 1 pages in unavailable ranges
[    0.016813] On node 0, zone DMA: 352 pages in unavailable ranges
[    0.033280] On node 0, zone DMA32: 20480 pages in unavailable ranges
[    0.033419] ACPI: LAPIC_NMI (acpi_id[0x01] dfl dfl lint[0x1])
[    0.055212] IOAPIC[0]: apic_id 4, version 17, address 0xfec00000, GSI 0-23
[    0.055240] ACPI: INT_SRC_OVR (bus 0 bus_irq 9 global_irq 9 high level)
[    0.055256] ACPI: Using ACPI (MADT) for SMP configuration information
[    0.055278] smpboot: Allowing 4 CPUs, 0 hotplug CPUs
[    0.055302] [mem 0x7b000000-0xffffffff] available for PCI devices
[    0.055306] Booting paravirtualized kernel on Hyper-V
[    0.055309] clocksource: refined-jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.068345] setup_percpu: NR_CPUS:256 nr_cpumask_bits:256 nr_cpu_ids:4 nr_node_ids:1
[    0.076050] percpu: Embedded 53 pages/cpu s179112 r8192 d29784 u524288
[    0.076076] pcpu-alloc: s179112 r8192 d29784 u524288 alloc=1*2097152
[    0.076084] pcpu-alloc: [0] 0 1 2 3
[    0.076116] Hyper-V: PV spinlocks enabled
[    0.076122] PV qspinlock hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.076132] Built 1 zonelists, mobility grouping on.  Total pages: 495429
[    0.076137] Kernel command line: initrd=\initrd.img WSL_ROOT_INIT=1 panic=-1 nr_cpus=4 swiotlb=force console=hvc0 debug pty.legacy_count=0
[    0.076357] Unknown kernel command line parameters "WSL_ROOT_INIT=1", will be passed to user space.
[    0.089653] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.090090] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.090359] mem auto-init: stack:all(zero), heap alloc:off, heap free:off
[    0.090381] software IO TLB: area num 4.
[    0.365753] Memory: 216940K/2013820K available (20498K kernel code, 2713K rwdata, 4232K rodata, 1840K init, 1740K bss, 146580K reserved, 0K cma-reserved)
[    0.365872] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.365895] ftrace: allocating 54704 entries in 214 pages
[    0.396138] ftrace: allocated 214 pages with 5 groups
[    0.408705] Dynamic Preempt: none
[    0.410716] rcu: Preemptible hierarchical RCU implementation.
[    0.410724] rcu:     RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=4.
[    0.410729]  Trampoline variant of Tasks RCU enabled.
[    0.410730]  Rude variant of Tasks RCU enabled.
[    0.410731]  Tracing variant of Tasks RCU enabled.
[    0.410733] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.410735] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.447640] Using NULL legacy PIC
[    0.447647] NR_IRQS: 16640, nr_irqs: 456, preallocated irqs: 0
[    0.447729] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.458424] random: crng init done
[    0.458515] Console: colour dummy device 80x25
[    0.458542] ACPI: Core revision 20220331
[    0.458961] Failed to register legacy timer interrupt
[    0.458966] APIC: Switch to symmetric I/O mode setup
[    0.458971] Hyper-V: enabling crash_kexec_post_notifiers
[    0.459038] Hyper-V: Using IPI hypercalls
[    0.459043] Hyper-V: Using enlightened APIC (xapic mode)
[    0.459450] Calibrating delay loop (skipped), value calculated using timer frequency.. 2188.79 BogoMIPS (lpj=10943990)
[    0.459459] pid_max: default: 32768 minimum: 301
[    0.459508] LSM: Security Framework initializing
[    0.459523] landlock: Up and running.
[    0.459585] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.459601] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.461112] Last level iTLB entries: 4KB 48, 2MB 0, 4MB 0
[    0.461126] Last level dTLB entries: 4KB 0, 2MB 0, 4MB 0, 1GB 0
[    0.461145] Spectre V1 : Mitigation: usercopy/swapgs barriers and __user pointer sanitization
[    0.461150] Spectre V2 : Mitigation: Retpolines
[    0.461151] Spectre V2 : Spectre v2 / SpectreRSB mitigation: Filling RSB on context switch
[    0.461153] Spectre V2 : Spectre v2 / SpectreRSB : Filling RSB on VMEXIT
[    0.461154] Spectre V2 : Enabling Restricted Speculation for firmware calls
[    0.461158] Spectre V2 : mitigation: Enabling conditional Indirect Branch Prediction Barrier
[    0.469437] Freeing SMP alternatives memory: 60K
[    0.469437] smpboot: CPU0: Intel(R) Pentium(R) CPU N4200 @ 1.10GHz (family: 0x6, model: 0x5c, stepping: 0x9)
[    0.469437] cblist_init_generic: Setting adjustable number of callback queues.
[    0.469437] cblist_init_generic: Setting shift to 2 and lim to 1.
[    0.469437] cblist_init_generic: Setting shift to 2 and lim to 1.
[    0.469437] cblist_init_generic: Setting shift to 2 and lim to 1.
[    0.469437] Performance Events: Goldmont events, 32-deep LBR, full-width counters, Intel PMU driver.
[    0.469437] ... version:                4
[    0.469437] ... bit width:              48
[    0.469437] ... generic registers:      4
[    0.469437] ... value mask:             0000ffffffffffff
[    0.469437] ... max period:             00007fffffffffff
[    0.469437] ... fixed-purpose events:   3
[    0.469437] ... event mask:             000000070000000f
[    0.469437] rcu: Hierarchical SRCU implementation.
[    0.469437] rcu:     Max phase no-delay instances is 1000.
[    0.470499] smp: Bringing up secondary CPUs ...
[    0.471248] x86: Booting SMP configuration:
[    0.471255] .... node  #0, CPUs:      #1 #2 #3
[    0.499820] smp: Brought up 1 node, 4 CPUs
[    0.499820] smpboot: Max logical packages: 1
[    0.499820] smpboot: Total of 4 processors activated (8755.19 BogoMIPS)
[    0.550704] node 0 deferred pages initialised in 40ms
[    0.550913] devtmpfs: initialized
[    0.550913] x86/mm: Memory block size: 128MB
[    0.550913] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.550913] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.551966] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.552522] thermal_sys: Registered thermal governor 'step_wise'
[    0.552700] cpuidle: using governor menu
[    0.552700] PCI: Fatal: No config space access function found
[    0.571326] kprobes: kprobe jump-optimization is enabled. All kprobes are optimized if possible.
[    0.586663] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    0.586663] HugeTLB: 16380 KiB vmemmap can be freed for a 1.00 GiB page
[    0.586663] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    0.586663] HugeTLB: 28 KiB vmemmap can be freed for a 2.00 MiB page
[    0.586663] raid6: skipped pq benchmark and selected sse2x4
[    0.586663] raid6: using ssse3x2 recovery algorithm
[    0.586663] ACPI: Added _OSI(Module Device)
[    0.586663] ACPI: Added _OSI(Processor Device)
[    0.586663] ACPI: Added _OSI(3.0 _SCP Extensions)
[    0.586663] ACPI: Added _OSI(Processor Aggregator Device)
[    0.586663] ACPI: Added _OSI(Linux-Dell-Video)
[    0.586663] ACPI: Added _OSI(Linux-Lenovo-NV-HDMI-Audio)
[    0.586663] ACPI: Added _OSI(Linux-HPI-Hybrid-Graphics)
[    0.605383] ACPI: 1 ACPI AML tables successfully acquired and loaded
[    0.653391] ACPI: Interpreter enabled
[    0.653391] ACPI: PM: (supports S0 S5)
[    0.653391] ACPI: Using IOAPIC for interrupt routing
[    0.653391] PCI: Using host bridge windows from ACPI; if necessary, use "pci=nocrs" and report a bug
[    0.653391] PCI: Using E820 reservations for host bridge windows
[    0.655781] ACPI: Enabled 2 GPEs in block 00 to 0F
[    0.675127] iommu: Default domain type: Translated
[    0.675135] iommu: DMA domain TLB invalidation policy: lazy mode
[    0.675376] SCSI subsystem initialized
[    0.675395] ACPI: bus type USB registered
[    0.675425] usbcore: registered new interface driver usbfs
[    0.675444] usbcore: registered new interface driver hub
[    0.675461] usbcore: registered new device driver usb
[    0.675493] pps_core: LinuxPPS API ver. 1 registered
[    0.675496] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.675501] PTP clock support registered
[    1.859815] hv_vmbus: Vmbus version:5.3
[    1.870052] PCI: Using ACPI for IRQ routing
[    1.870088] PCI: System does not support PCI
[    1.870668] clocksource: Switched to clocksource hyperv_clocksource_tsc_page
[    1.870668] VFS: Disk quotas dquot_6.6.0
[    1.870668] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    1.870668] FS-Cache: Loaded
[    1.870668] pnp: PnP ACPI init
[    1.875324] pnp: PnP ACPI: found 1 devices
[    1.906770] NET: Registered PF_INET protocol family
[    1.909718] IP idents hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    1.911009] tcp_listen_portaddr_hash hash table entries: 1024 (order: 2, 16384 bytes, linear)
[    1.911033] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    1.911040] TCP established hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    1.911116] TCP bind hash table entries: 16384 (order: 6, 262144 bytes, linear)
[    1.911244] TCP: Hash tables configured (established 16384 bind 16384)
[    1.911360] UDP hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    1.911397] UDP-Lite hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    1.911504] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    1.912781] RPC: Registered named UNIX socket transport module.
[    1.912790] RPC: Registered udp transport module.
[    1.912791] RPC: Registered tcp transport module.
[    1.912793] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    1.912807] PCI: CLS 0 bytes, default 64
[    1.913153] KVM: vmx: using Hyper-V Enlightened VMCS
[    1.913163] Trying to unpack rootfs image as initramfs...
[    1.929426] Freeing initrd memory: 1888K
[    2.261218] kvm: already loaded vendor module 'kvm_intel'
[    2.268119] Initialise system trusted keyrings
[    2.276386] workingset: timestamp_bits=46 max_order=19 bucket_order=0
[    2.278746] zbud: loaded
[    2.280192] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    2.281193] NFS: Registering the id_resolver key type
[    2.281220] Key type id_resolver registered
[    2.281222] Key type id_legacy registered
[    2.281227] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    2.281235] nfs4flexfilelayout_init: NFSv4 Flexfile Layout Driver Registering...
[    2.287971] Key type cifs.idmap registered
[    2.288376] fuse: init (API version 7.36)
[    2.289217] SGI XFS with ACLs, security attributes, realtime, scrub, repair, quota, no debug enabled
[    2.291074] 9p: Installing v9fs 9p2000 file system support
[    2.291242] ceph: loaded (mds proto 32)
[    2.330125] NET: Registered PF_ALG protocol family
[    2.330159] xor: measuring software checksum speed
[    2.331670]    prefetch64-sse  :  8374 MB/sec
[    2.333051]    generic_sse     :  7236 MB/sec
[    2.333054] xor: using function: prefetch64-sse (8374 MB/sec)
[    2.333071] Key type asymmetric registered
[    2.333074] Asymmetric key parser 'x509' registered
[    2.333165] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 248)
[    2.335074] hv_vmbus: registering driver hv_pci
[    2.356127] hv_pci 35453a7c-0ce2-48aa-a318-ad84e3e938e9: PCI VMBus probing: Using version 0x10004
[    2.377224] hv_pci 35453a7c-0ce2-48aa-a318-ad84e3e938e9: PCI host bridge to bus 0ce2:00
[    2.377234] pci_bus 0ce2:00: root bus resource [mem 0x9ffe00000-0x9ffe02fff window]
[    2.377240] pci_bus 0ce2:00: No busn resource found for root bus, will use [bus 00-ff]
[    2.404743] pci 0ce2:00:00.0: [1af4:1043] type 00 class 0x010000
[    2.411484] pci 0ce2:00:00.0: reg 0x10: [mem 0x9ffe00000-0x9ffe00fff 64bit]
[    2.417681] pci 0ce2:00:00.0: reg 0x18: [mem 0x9ffe01000-0x9ffe01fff 64bit]
[    2.484482] pci 0ce2:00:00.0: reg 0x20: [mem 0x9ffe02000-0x9ffe02fff 64bit]
[    2.535336] pci_bus 0ce2:00: busn_res: [bus 00-ff] end is updated to 00
[    2.535359] pci 0ce2:00:00.0: BAR 0: assigned [mem 0x9ffe00000-0x9ffe00fff 64bit]
[    2.537763] pci 0ce2:00:00.0: BAR 2: assigned [mem 0x9ffe01000-0x9ffe01fff 64bit]
[    2.539169] pci 0ce2:00:00.0: BAR 4: assigned [mem 0x9ffe02000-0x9ffe02fff 64bit]
[    2.545366] hv_pci f865f256-d884-4bc4-8ec1-12539e2dab06: PCI VMBus probing: Using version 0x10004
[    2.595061] hv_pci f865f256-d884-4bc4-8ec1-12539e2dab06: PCI host bridge to bus d884:00
[    2.595071] pci_bus d884:00: root bus resource [mem 0x9ffe04000-0x9ffe06fff window]
[    2.595077] pci_bus d884:00: No busn resource found for root bus, will use [bus 00-ff]
[    2.644405] pci d884:00:00.0: [1af4:1049] type 00 class 0x010000
[    2.653891] pci d884:00:00.0: reg 0x10: [mem 0x9ffe04000-0x9ffe04fff 64bit]
[    2.657866] pci d884:00:00.0: reg 0x18: [mem 0x9ffe05000-0x9ffe05fff 64bit]
[    2.691267] pci d884:00:00.0: reg 0x20: [mem 0x9ffe06000-0x9ffe06fff 64bit]
[    2.839725] pci_bus d884:00: busn_res: [bus 00-ff] end is updated to 00
[    2.839747] pci d884:00:00.0: BAR 0: assigned [mem 0x9ffe04000-0x9ffe04fff 64bit]
[    2.841197] pci d884:00:00.0: BAR 2: assigned [mem 0x9ffe05000-0x9ffe05fff 64bit]
[    2.842495] pci d884:00:00.0: BAR 4: assigned [mem 0x9ffe06000-0x9ffe06fff 64bit]
[    2.845221] ACPI: AC: AC Adapter [AC1] (on-line)
[    2.855034] virtio-pci 0ce2:00:00.0: enabling device (0000 -> 0002)
[    2.861286] ACPI: battery: Slot [BAT1] (battery present)
[    2.898648] virtio-pci d884:00:00.0: enabling device (0000 -> 0002)
[    2.921988] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
[    3.218984] Non-volatile memory driver v1.3
[    3.219017] ACPI: bus type drm_connector registered
[    3.220566] [drm] Initialized vgem 1.0.0 20120112 for vgem on minor 0
[    3.225677] printk: console [hvc0] enabled
[    3.240895] brd: module loaded
[    3.245004] loop: module loaded
[    3.247758] zram: Added device: zram0
[    3.251143] hv_vmbus: registering driver hv_storvsc
[    3.254711] wireguard: WireGuard 1.0.0 loaded. See www.wireguard.com for information.
[    3.259190] scsi host0: storvsc_host_t
[    3.269701] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[    3.269799] tun: Universal TUN/TAP device driver, 1.6
[    3.270139] PPP generic driver version 2.4.2
[    3.282250] PPP BSD Compression module registered
[    3.287745] PPP Deflate Compression module registered
[    3.297172] PPP MPPE Compression module registered
[    3.304779] NET: Registered PF_PPPOX protocol family
[    3.307688] usbcore: registered new interface driver cdc_ether
[    3.310666] usbcore: registered new interface driver cdc_ncm
[    3.314554] usbcore: registered new interface driver r8153_ecm
[    3.318346] hv_vmbus: registering driver hv_netvsc
[    3.321312] VFIO - User Level meta-driver version: 0.3
[    3.333359] usbcore: registered new interface driver cdc_acm
[    3.336386] cdc_acm: USB Abstract Control Model driver for USB modems and ISDN adapters
[    3.340917] usbcore: registered new interface driver ch341
[    3.343687] usbserial: USB Serial support registered for ch341-uart
[    3.352085] usbcore: registered new interface driver cp210x
[    3.354683] usbserial: USB Serial support registered for cp210x
[    3.358441] usbcore: registered new interface driver ftdi_sio
[    3.371194] usbserial: USB Serial support registered for FTDI USB Serial Device
[    3.374251] vhci_hcd vhci_hcd.0: USB/IP Virtual Host Controller
[    3.376505] vhci_hcd vhci_hcd.0: new USB bus registered, assigned bus number 1
[    3.379288] vhci_hcd: created sysfs vhci_hcd.0
[    3.381898] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.00
[    3.385286] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.387361] usb usb1: Product: USB/IP Virtual Host Controller
[    3.389767] usb usb1: Manufacturer: Linux 6.0.1-microsoft-quark-WSL2 vhci_hcd
[    3.395232] usb usb1: SerialNumber: vhci_hcd.0
[    3.399361] hub 1-0:1.0: USB hub found
[    3.401223] hub 1-0:1.0: 8 ports detected
[    3.403064] vhci_hcd vhci_hcd.0: USB/IP Virtual Host Controller
[    3.405240] vhci_hcd vhci_hcd.0: new USB bus registered, assigned bus number 2
[    3.407410] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    3.410341] usb usb2: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 6.00
[    3.414225] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.418197] usb usb2: Product: USB/IP Virtual Host Controller
[    3.422776] usb usb2: Manufacturer: Linux 6.0.1-microsoft-quark-WSL2 vhci_hcd
[    3.436421] usb usb2: SerialNumber: vhci_hcd.0
[    3.439761] hub 2-0:1.0: USB hub found
[    3.452141] hub 2-0:1.0: 8 ports detected
[    3.458091] hv_vmbus: registering driver hyperv_keyboard
[    3.465072] rtc_cmos 00:00: RTC can wake from S4
[    3.474001] rtc_cmos 00:00: registered as rtc0
[    3.476507] rtc_cmos 00:00: setting system clock to 2022-10-18T19:57:27 UTC (1666123047)
[    3.483253] rtc_cmos 00:00: alarms up to one month, 114 bytes nvram
[    3.489237] device-mapper: ioctl: 4.47.0-ioctl (2022-07-28) initialised: dm-devel@redhat.com
[    3.493767] device-mapper: raid: Loading target version 1.15.1
[    3.498017] usbcore: registered new interface driver usbhid
[    3.501125] usbhid: USB HID core driver
[    3.502855] hv_utils: Registering HyperV Utility Driver
[    3.505702] hv_vmbus: registering driver hv_utils
[    3.509708] hv_vmbus: registering driver hv_balloon
[    3.521435] hv_vmbus: registering driver dxgkrnl
[    3.525803] hv_utils: TimeSync IC version 4.0
[    3.532277] drop_monitor: Initializing network drop monitor service
[    3.538876] hv_balloon: Using Dynamic Memory protocol version 2.0
[    3.538937] Mirror/redirect action on
[    3.542210] u32 classifier
[    3.543298]     Performance counters on
[    3.548466]     input device check on
[    3.550449]     Actions configured
[    3.552723] Free page reporting enabled
[    3.554297] hv_balloon: Cold memory discard hint enabled
[    3.555002] IPVS: Registered protocols (TCP, UDP)
[    3.557486] IPVS: Connection hash table configured (size=4096, memory=32Kbytes)
[    3.559698] IPVS: ipvs loaded.
[    3.560812] IPVS: [rr] scheduler registered.
[    3.562783] IPVS: [wrr] scheduler registered.
[    3.565608] IPVS: [sh] scheduler registered.
[    3.567728] ipip: IPv4 and MPLS over IPv4 tunneling driver
[    3.570244] ipt_CLUSTERIP: ClusterIP Version 0.8 loaded successfully
[    3.592784] Initializing XFRM netlink socket
[    3.603894] NET: Registered PF_INET6 protocol family
[    3.608286] Segment Routing with IPv6
[    3.616741] In-situ OAM (IOAM) with IPv6
[    3.622610] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[    3.641618] NET: Registered PF_PACKET protocol family
[    3.655448] Bridge firewalling registered
[    3.659088] 8021q: 802.1Q VLAN Support v1.8
[    3.668117] sctp: Hash tables configured (bind 256/256)
[    3.678410] 9pnet: Installing 9P2000 support
[    4.121410] Key type dns_resolver registered
[    4.152315] Key type ceph registered
[    4.183291] libceph: loaded (mon/osd proto 15/24)
[    4.208915] NET: Registered PF_VSOCK protocol family
[    4.225569] hv_vmbus: registering driver hv_sock
[    4.238851] IPI shorthand broadcast: enabled
[    4.242413] sched_clock: Marking stable (4241215700, 1090100)->(4412888500, -170582700)
[    4.262288] registered taskstats version 1
[    4.273860] Loading compiled-in X.509 certificates
[    4.275927] zswap: loaded using pool lzo/zbud
[    4.283171] Btrfs loaded, crc32c=crc32c-generic, zoned=no, fsverity=no
[    4.312012] Freeing unused kernel image (initmem) memory: 1840K
[    4.359921] Write protecting the kernel read-only data: 28672k
[    4.380440] Freeing unused kernel image (text/rodata gap) memory: 2028K
[    4.400378] Freeing unused kernel image (rodata/data gap) memory: 1912K
[    4.407721] Run /init as init process
[    4.409680]   with arguments:
[    4.416351]     /init
[    4.417490]   with environment:
[    4.420355]     HOME=/
[    4.422375]     TERM=linux
[    4.424094]     WSL_ROOT_INIT=1
[    4.455055] scsi 0:0:0:0: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
[    4.468412] sd 0:0:0:0: Attached scsi generic sg0 type 0
[    4.486470] sd 0:0:0:0: [sda] 748624 512-byte logical blocks: (383 MB/366 MiB)
[    4.491591] sd 0:0:0:0: [sda] Write Protect is on
[    4.494004] sd 0:0:0:0: [sda] Mode Sense: 0f 00 80 00
[    4.511984] sd 0:0:0:0: [sda] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
[    4.810723] sd 0:0:0:0: [sda] Attached SCSI disk
[    4.931186] scsi 0:0:0:1: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
[    4.932912] EXT4-fs (sda): mounted filesystem without journal. Quota mode: none.
[    4.970842] scsi 0:0:0:1: Attached scsi generic sg1 type 0
[    5.005723] sd 0:0:0:1: [sdb] 2097160 512-byte logical blocks: (1.07 GB/1.00 GiB)
[    5.016474] sd 0:0:0:1: [sdb] 4096-byte physical blocks
[    5.024710] sd 0:0:0:1: [sdb] Write Protect is off
[    5.026464] sd 0:0:0:1: [sdb] Mode Sense: 0f 00 00 00
[    5.032478] sd 0:0:0:1: [sdb] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    5.052150] sd 0:0:0:1: [sdb] Attached SCSI disk
[    5.135699] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    5.631465] Adding 1048576k swap on /dev/sdb.  Priority:-2 extents:1 across:1048576k FS
[    5.835085] hv_pci 064f1052-2bdb-4264-a9ba-144a777305ab: PCI VMBus probing: Using version 0x10004
[    5.911509] hv_pci 064f1052-2bdb-4264-a9ba-144a777305ab: PCI host bridge to bus 2bdb:00
[    5.971446] pci_bus 2bdb:00: root bus resource [mem 0xa00000000-0xc00001fff window]
[    5.999475] pci_bus 2bdb:00: No busn resource found for root bus, will use [bus 00-ff]
[    6.007034] pci 2bdb:00:00.0: [1af4:105a] type 00 class 0x088000
[    6.034438] pci 2bdb:00:00.0: reg 0x10: [mem 0xc00000000-0xc00000fff 64bit]
[    6.038386] pci 2bdb:00:00.0: reg 0x18: [mem 0xc00001000-0xc00001fff 64bit]
[    6.100600] pci 2bdb:00:00.0: reg 0x20: [mem 0xa00000000-0xbffffffff 64bit]
[    6.202193] pci_bus 2bdb:00: busn_res: [bus 00-ff] end is updated to 00
[    6.205465] pci 2bdb:00:00.0: BAR 4: assigned [mem 0xa00000000-0xbffffffff 64bit]
[    6.227848] pci 2bdb:00:00.0: BAR 0: assigned [mem 0xc00000000-0xc00000fff 64bit]
[    6.237608] pci 2bdb:00:00.0: BAR 2: assigned [mem 0xc00001000-0xc00001fff 64bit]
[    6.242884] virtio-pci 2bdb:00:00.0: enabling device (0000 -> 0002)
[    6.352374] virtiofs virtio2: Cache len: 0x200000000 @ 0xa00000000
[    6.937023] memmap_init_zone_device initialised 2097152 pages in 540ms
[    7.808796] scsi 0:0:0:2: Direct-Access     Msft     Virtual Disk     1.0  PQ: 0 ANSI: 5
[    7.908507] sd 0:0:0:2: Attached scsi generic sg2 type 0
[    7.927954] sd 0:0:0:2: [sdc] 536870912 512-byte logical blocks: (275 GB/256 GiB)
[    7.955968] sd 0:0:0:2: [sdc] 4096-byte physical blocks
[    8.017284] sd 0:0:0:2: [sdc] Write Protect is off
[    8.085794] sd 0:0:0:2: [sdc] Mode Sense: 0f 00 00 00
[    8.173781] sd 0:0:0:2: [sdc] Write cache: enabled, read cache: enabled, doesn't support DPO or FUA
[    8.498655] sd 0:0:0:2: [sdc] Attached SCSI disk
[    8.748251] EXT4-fs (sdc): recovery complete
[    8.814692] EXT4-fs (sdc): mounted filesystem with ordered data mode. Quota mode: none.
[   11.619350] hv_pci b573bdc6-03aa-4499-b295-b5b92415d314: PCI VMBus probing: Using version 0x10004
[   11.631843] 9pnet_virtio: no channels available for device drvfsa
[   11.783667] WSL (1) WARNING: mount: waiting for virtio device drvfsa
[   11.854349] hv_pci b573bdc6-03aa-4499-b295-b5b92415d314: PCI host bridge to bus 03aa:00
[   11.913190] pci_bus 03aa:00: root bus resource [mem 0x9ffe08000-0x9ffe0afff window]
[   11.951076] 9pnet_virtio: no channels available for device drvfsa
[   11.963046] pci_bus 03aa:00: No busn resource found for root bus, will use [bus 00-ff]
[   12.332523] 9pnet_virtio: no channels available for device drvfsa
[   12.516089] 9pnet_virtio: no channels available for device drvfsa
[   12.574009] pci 03aa:00:00.0: [1af4:1049] type 00 class 0x010000
[   12.671427] 9pnet_virtio: no channels available for device drvfsa
[   12.756254] pci 03aa:00:00.0: reg 0x10: [mem 0x9ffe08000-0x9ffe08fff 64bit]
[   12.773140] pci 03aa:00:00.0: reg 0x18: [mem 0x9ffe09000-0x9ffe09fff 64bit]
[   12.807826] 9pnet_virtio: no channels available for device drvfsa
[   12.934250] pci 03aa:00:00.0: reg 0x20: [mem 0x9ffe0a000-0x9ffe0afff 64bit]
[   12.968470] 9pnet_virtio: no channels available for device drvfsa
[   13.085241] 9pnet_virtio: no channels available for device drvfsa
[   13.223338] 9pnet_virtio: no channels available for device drvfsa
[   13.387374] 9pnet_virtio: no channels available for device drvfsa
[   13.506449] 9pnet_virtio: no channels available for device drvfsa
[   13.635949] 9pnet_virtio: no channels available for device drvfsa
[   13.725511] WSL (1) ERROR: MountWithRetry:306: mount(drvfsa, /mnt/d, 9p, 0x00000400, cache=mmap,msize=262144,trans=virtio,aname=drvfs;path=D:\;uid=1000;gid=1000;symlinkroot=/mnt/
[   13.725523] ) failed: Resource busy
[   13.957283] pci_bus 03aa:00: busn_res: [bus 00-ff] end is updated to 00
[   13.982051] pci 03aa:00:00.0: BAR 0: assigned [mem 0x9ffe08000-0x9ffe08fff 64bit]
[   14.066826] pci 03aa:00:00.0: BAR 2: assigned [mem 0x9ffe09000-0x9ffe09fff 64bit]
[   14.118341] pci 03aa:00:00.0: BAR 4: assigned [mem 0x9ffe0a000-0x9ffe0afff 64bit]
[   14.157600] virtio-pci 03aa:00:00.0: enabling device (0000 -> 0002)
[   15.783517] hv_pci 6b198b83-7726-4b06-924e-0982730046a2: PCI VMBus probing: Using version 0x10004
[   15.791903] 9pnet_virtio: no channels available for device drvfsa
[   15.913524] WSL (1) WARNING: mount: waiting for virtio device drvfsa
[   15.972557] hv_pci 6b198b83-7726-4b06-924e-0982730046a2: PCI host bridge to bus 7726:00
[   16.019101] pci_bus 7726:00: root bus resource [mem 0x9ffe0c000-0x9ffe0efff window]
[   16.035925] 9pnet_virtio: no channels available for device drvfsa
[   16.050921] pci_bus 7726:00: No busn resource found for root bus, will use [bus 00-ff]
[   16.291239] 9pnet_virtio: no channels available for device drvfsa
[   16.559590] 9pnet_virtio: no channels available for device drvfsa
[   16.655713] pci 7726:00:00.0: [1af4:1049] type 00 class 0x010000
[   16.729386] 9pnet_virtio: no channels available for device drvfsa
[   16.883725] 9pnet_virtio: no channels available for device drvfsa
[   17.074469] 9pnet_virtio: no channels available for device drvfsa
[   17.236640] pci 7726:00:00.0: reg 0x10: [mem 0x9ffe0c000-0x9ffe0cfff 64bit]
[   17.280693] 9pnet_virtio: no channels available for device drvfsa
[   17.536098] 9pnet_virtio: no channels available for device drvfsa
[   17.633021] pci 7726:00:00.0: reg 0x18: [mem 0x9ffe0d000-0x9ffe0dfff 64bit]
[   17.704982] 9pnet_virtio: no channels available for device drvfsa
[   17.885206] WSL (1) ERROR: MountWithRetry:306: mount(drvfsa, /mnt/d, 9p, 0x00000400, cache=mmap,msize=262144,trans=virtio,aname=drvfs;path=D:\;uid=1000;gid=1000;symlinkroot=/mnt/
[   17.885221] ) failed: Resource busy
[   18.128131] pci 7726:00:00.0: reg 0x20: [mem 0x9ffe0e000-0x9ffe0efff 64bit]
[   18.782056] pci_bus 7726:00: busn_res: [bus 00-ff] end is updated to 00
[   18.803457] pci 7726:00:00.0: BAR 0: assigned [mem 0x9ffe0c000-0x9ffe0cfff 64bit]
[   18.919653] pci 7726:00:00.0: BAR 2: assigned [mem 0x9ffe0d000-0x9ffe0dfff 64bit]
[   19.033876] pci 7726:00:00.0: BAR 4: assigned [mem 0x9ffe0e000-0x9ffe0efff 64bit]
[   19.330858] virtio-pci 7726:00:00.0: enabling device (0000 -> 0002)
[   21.110151] misc dxg: dxgk: dxgglobal_acquire_channel_lock: Failed to acquire global channel lock
```

I see the same behavior with default kernel either

**Comment by benhillis** [comment]
I think the dxg print may not be related to why the PCI devices cannot be accessed. Would need to look at logs of what's happening on the host:

/logs

**Comment by Sarisan** [comment]
[WslLogs-2022-10-19_01-12-22.zip](https://github.com/microsoft/WSL/files/9814799/WslLogs-2022-10-19_01-12-22.zip)

**Comment by dfisk** [comment]
I am getting the same dmesg dgx print error on Windows Server 2022, version 21H2 build 203487.1364. It is the last entry after Ubuntu install and occurs after every invocation of nvidia-smi  (post installing CUDA 12 with a GPU). The nvidia-command returns:
Failed to initialize NVML: GPU access blocked by the operating system
Failed to properly shut down NVML: GPU access blocked by the operating system

**Comment by dfisk** [comment]
This Event looks relevant:
Failed to map guest I/O buffer for read access with status 0xC00000A1. Device name = C:\Users\Administrator\AppData\Local\Packages\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc\LocalState\ext4.vhdx

**Comment by geole** [comment]
Hello, i have probably the same problem
This is my wsl collect
[WslLogs-2023-03-07_19-11-24.zip](https://github.com/microsoft/WSL/files/10912646/WslLogs-2023-03-07_19-11-24.zip)

i have also a
[Journal.txt](https://github.com/microsoft/WSL/files/10912773/Journal.txt)
  ubuntu   journalog
I can do another test.

**Comment by geole** [comment]
Hello
Last Errors
a@DESKTOP-71NRFPV:~$ journalctl -p err -n 5
Mar 10 13:43:38 DESKTOP-71NRFPV kernel: misc dxg: dxgk: dxgglobal_acquire_channel_lock: Failed to acquire global channel lock
Mar 10 13:55:40 DESKTOP-71NRFPV kernel: misc dxg: dxgk: dxgglobal_acquire_channel_lock: Failed to acquire global channel lock
Mar 10 13:55:41 DESKTOP-71NRFPV gnome-session-binary[11450]: Unrecoverable failure in required component org.gnome.Shell.desktop
Mar 10 14:00:14 DESKTOP-71NRFPV kernel: misc dxg: dxgk: dxgglobal_acquire_channel_lock: Failed to acquire global channel lock
Mar 10 14:00:14 DESKTOP-71NRFPV gnome-session-binary[11843]: Unrecoverable failure in required component org.gnome.Shell.desktop
a@DESKTOP-71NRFPV:~$

Journal trace of last connexion
[Trace.txt](https://github.com/microsoft/WSL/files/10942619/Trace.txt)

**Comment by wizpresso-steve-cy-fan** [comment]
We also saw this issue popping off for #10357. Consider the relevant parts of kernel driver:
```c
	down_read(&dxgglobal->channel_lock);
	if (dxgglobal->channel.channel == NULL) {
		DXG_ERR("Failed to acquire global channel lock");
		return -ENODEV;
	} else {
		return 0;
	}
```
https://github.com/microsoft/WSL2-Linux-Kernel/blob/12802bc3d28e15a13881c8edaef28238fd44c90d/drivers/hv/dxgkrnl/dxgmodule.c#L49C2-L55

It seems like `dxgglobal->channel.channel`. This means we don't even have a dxgkrnl IPC channel from Hyper-V.

**Comment by Sarisan** [comment]
https://github.com/microsoft/WSL2-Linux-Kernel/commit/423e5044c57566f8267bbb20cf7cf57d1aefb040
